nntrainer/layers/preprocess_translate_layer.cpp

```diff
@@ -45,75 +47,100 @@ int PreprocessTranslateLayer::initialize(Manager &manager) {
     affine_transform_mat.at<float>(1, 1) = 1;
 
     input_mat =
-      cv::Mat::zeros(input_dim[0].height(), input_dim[0].width(), CV_32FC3);
+      cv::Mat::zeros(input_dim_0.height(), input_dim_0.width(), CV_32FC3);
     output_mat =
-      cv::Mat::zeros(input_dim[0].height(), input_dim[0].width(), CV_32FC3);
+      cv::Mat::zeros(input_dim_0.height(), input_dim_0.width(), CV_32FC3);
 #else
     throw exception::not_supported(
       "Preprocess translate layer is not supported without opencv");
 #endif
   }
+}
+
+void PreprocessTranslateLayer::setProperty(
+  const std::vector<std::string> &values) {
+  /// @todo: deprecate this in favor of loadProperties
+  for (unsigned int i = 0; i < values.size(); ++i) {
+    std::string key;
+    std::string value;
+    std::stringstream ss;
+
+    if (getKeyValue(values[i], key, value) != ML_ERROR_NONE) {
+      throw std::invalid_argument("Error parsing the property: " + values[i]);
+    }
 
-  return ML_ERROR_NONE;
+    if (value.empty()) {
+      ss << "value is empty: key: " << key << ", value: " << value;
+      throw std::invalid_argument(ss.str());
+    }
+
+    /// @note this calls derived setProperty if available
+    setProperty(key, value);
+  }
 }
 
-void PreprocessTranslateLayer::setProperty(const PropertyType type,
+void PreprocessTranslateLayer::setProperty(const std::string &type_str,
                                            const std::string &value) {
+  using PropertyType = LayerV1::PropertyType;
   int status = ML_ERROR_NONE;
+  LayerV1::PropertyType type =
+    static_cast<LayerV1::PropertyType>(parseLayerProperty(type_str));
 
   switch (type) {
-  case PropertyType::random_translate:
-    if (!value.empty()) {
-      status = setFloat(translation_factor, value);
-      translation_factor = std::abs(translation_factor);
-      throw_status(status);
-    }
-    break;
+  case PropertyType::random_translate: {
+    status = setFloat(translation_factor, value);
+    translation_factor = std::abs(translation_factor);
+    throw_status(status);
+  } break;
   default:
-    LayerV1::setProperty(type, value);
-    break;
+    std::string msg =
+      "[PreprocessTranslateLayer] Unknown Layer Property Key for value " +
+      std::string(value);
+    throw exception::not_supported(msg);
   }
 }
 
-void PreprocessTranslateLayer::forwarding(bool training) {
+void PreprocessTranslateLayer::forwarding(RunLayerContext &context,
+                                          bool training) {
   if (!training) {
-    for (unsigned int idx = 0; idx < input_dim.size(); idx++) {
-      net_hidden[idx]->getVariableRef() = net_input[idx]->getVariableRef();
+    for (unsigned int idx = 0; idx < context.getNumInputs(); idx++) {
+      context.getOutput(idx) = context.getInput(idx);
```

Comment:
same for the record

Reply:
same reply as https://github.com/nnstreamer/nntrainer/pull/1386#discussion_r669254383